search/tasks.py

```diff
@@ -0,0 +1,108 @@
+from celery import shared_task
+from django.db.utils import DatabaseError
+from elasticsearch.exceptions import TransportError
+from elasticsearch.helpers import bulk
+from elasticsearch_dsl.connections import connections
+from dips.models import Collection, DIP, DigitalFile
+
+import logging
+
+# Use a normal logger to avoid redirecting both `stdout` and `stderr` to the
+# logger and back when using Celery's `get_task_logger`, and to avoid changing
+# the default `CELERY_REDIRECT_STDOUTS_LEVEL` when using `print`.
+logger = logging.getLogger("search.tasks")
+
+
+@shared_task(
+    autoretry_for=(TransportError, DatabaseError),
+    max_retries=10,
+    default_retry_delay=30,
+    ignore_result=True,
+)
+def update_es_descendants(class_name, pk):
+    """Update the related DigitalFiles documents in ES.
+
+    With the partial data from the ancestor Collection or DIP.
+    """
+    if class_name not in ["Collection", "DIP"]:
+        raise Exception("Can not update descendants of %s." % class_name)
+    logger.info("Updating DigitalFiles of %s [id: %s] " % (class_name, pk))
+    if class_name == "Collection":
+        collection = Collection.objects.get(pk=pk)
+        # Partial update with `doc` doesn't remove the fields missing in data,
+        # they have to be removed via script to clear the existing value,
+        # `script` and `doc` can't be combined in update actions, therefore
+        # it's required to generate a Painless script to perform the update.
+        script = {
+            "source": "ctx._source.collection = params.collection;",
+            "lang": "painless",
+            "params": {"collection": collection.get_es_data_for_files()},
+        }
+        files = DigitalFile.objects.filter(dip__collection__pk=pk).all()
```

Comment:
I think you can use [.values_list('id', flat=True)](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#values-list) here to save loading unnecessary fields.